### PR TITLE
chore: tuck rollback sync

### PR DESCRIPTION
**Pull request by pepyakin**
This is the last commit on my quest of splitting up the sync module and isolating each individual component.

Closes THR-33

### CODE REVIEW

**Comment by pepyakin** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/thrumdev/nomt/584?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#584** <a href="https://app.graphite.dev/github/pr/thrumdev/nomt/584?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/thrumdev/nomt/584?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#583** <a href="https://app.graphite.dev/github/pr/thrumdev/nomt/583?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#582** <a href="https://app.graphite.dev/github/pr/thrumdev/nomt/582?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#581** <a href="https://app.graphite.dev/github/pr/thrumdev/nomt/581?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#577** <a href="https://app.graphite.dev/github/pr/thrumdev/nomt/577?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#576** <a href="https://app.graphite.dev/github/pr/thrumdev/nomt/576?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#575** <a href="https://app.graphite.dev/github/pr/thrumdev/nomt/575?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: nomt/src/store/sync.rs [hunk 1]
```diff
@@ -53,29 +47,18 @@ impl Sync {
...
         }
 
         // TODO: comprehensive error handling is coming later.
         bitbox_sync.wait_pre_meta().unwrap();
         let beatree_meta_wd = beatree_sync.wait_pre_meta().unwrap();
+        let (rollback_start_live, rollback_end_live) = if let Some(ref rollback) = rollback_sync {
+            rollback.wait_pre_meta()
+        } else {
+            (0, 0)
+        };
```
Comment (gabriele-0201) [inline nomt/src/store/sync.rs:61]: this chain of `=` is a little bit hard to follow. Not pushing for a change, just another option that is, just for me, more readable

```rust
        let (rollback_start_live, rollback_end_live) = match rollback_sync {
            Some(ref rollback) => rollback.wait_pre_meta(),
            None => (0, 0)
        }; 
```

**Review by gabriele-0201 — APPROVED** [review]
Now after THR-33 `Sync::sync(...)` is 63 lines of pure simplicity!

File: nomt/src/store/sync.rs (hunk 1 above)
Comment (pepyakin) [reply nomt/src/store/sync.rs:64]: agreed

**Comment by pepyakin** [comment]
### Merge activity

* **Nov 29, 2:35 PM EST**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/thrumdev/nomt/584).
